### ISSUE TITLE
[Faceted Filters] Price filter input missing accurate accessible name

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#113]
## Issue summary

The Price facet `input` elements do not feature a programmatically associated `label` to describe their purpose. Currently the `placeholder` attribute is used as the accessible name. As a result, there is an incorrect description announced on focus. This could lead to a confusing or frustrating user experience for screen reader or voice dictation users who rely on accurate labelling.

<details>
  <summary>Screenshots</summary>
<img width="488" alt="Screen shot of faceted filters section. Price filter is expanded, input controls are highlighted. " src="https://user-images.githubusercontent.com/1392632/124607175-976ef480-de3b-11eb-89d1-072b766233a7.png">

</details>

#### HTML

```html
<input
  class="field__input"
  name="filter.v.price.gte"
  id="Filter-Price-4"
  type="number"
  placeholder="0"
  min="0"
  max="615.00"
/>
<label
  class="field__label"
  for="Search-In-Modal"
>
  From
</label>

<!-- … -->

<input
  class="field__input"
  name="filter.v.price.lte"
  id="Filter-Price-4"
  type="number"
  placeholder="615.00"
  min="0"
  max="615.00"
/>
<label
  class="field__label"
  for="Search-In-Modal"
>
  To
</label>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported browser.
2. Activate the Price facet filter control.
3. With an active screen reader, navigate to the `input` controls.

### Expected

- `input` to be accompanied by a descriptive accessible name.

### Actual

- `input` accessible name/`label` association are inaccurate.

## Recommendation

1. Supply a unique `id` for each `input` control.
2. Adjust the `for` attribute value to match the `id` of the associated `input`.

#### HTML

```html
<input
  class="field__input"
  name="filter.v.price.gte"
  id="Filter-Price-GTE"
  type="number"
  placeholder="0"
  min="0"
  max="615.00"
/>
<label
  class="field__label"
  for="Filter-Price-GTE"
>
  From
</label>

<!-- … -->

<input
  class="field__input"
  name="filter.v.price.lte"
  id="Filter-Price-LTE"
  type="number"
  placeholder="615.00"
  min="0"
  max="615.00"
/>
<label
  class="field__label"
  for="Filter-Price-LTE"
>
  To
</label>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Understandable
- **WCAG SC:** [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- **Severity:** High
- **Effort:** Low
<!-- - **Related:** -->
